### PR TITLE
[MU4] Fix (MSVC/MinGW) compiler warnings

### PR DESCRIPTION
**Pull request by Jojo-Schmitz**

### CODE REVIEW

File: src/notation/notationerrors.h [hunk 1]
```diff
@@ -98,14 +98,12 @@ inline Ret make_ret(Err err, const io::path& filePath = "")
                                 "You can convert this score by opening and then\n"
                                 "saving with MuseScore version 2.x.\n"
                                 "Visit the %1MuseScore download page%2 to obtain such a 2.x version.")
-               .arg("<a href=\"https://musescore.org/download#older-versions\">")
-               .arg("</a>");
+               .arg("<a href=\"https://musescore.org/download#older-versions\">", "</a>");
```
Comment (Jojo-Schmitz) [inline src/notation/notationerrors.h:101]: not really a compiler wartning, but rather a hint from InteliSense(?)

File: src/notation/notationerrors.h [hunk 2]
```diff
@@ -129,6 +127,7 @@ inline Ret make_ret(Err err, const io::path& filePath = "")
     case Err::UnknownError:
     case Err::UserAbort:
     case Err::IgnoreError:
+    case Err::EmptySelection:
```
Comment (Jojo-Schmitz) [inline src/notation/notationerrors.h:130]: Do we want to ignore this (new) 'error'?

File: src/framework/mpe/mpetypes.h [hunk 3]
```diff
@@ -625,8 +625,8 @@ struct ArticulationMap : public SharedHashMap<ArticulationType, ArticulationAppl
         m_averagePitchRange = 0;
 
         for (size_t i = 0; i < EXPECTED_SIZE; ++i) {
-            m_averagePitchOffsetMap.emplace(i * TEN_PERCENT, 0);
-            m_averageDynamicOffsetMap.emplace(i * TEN_PERCENT, 0);
+            m_averagePitchOffsetMap.emplace(static_cast<int>(i) * TEN_PERCENT, 0);
+            m_averageDynamicOffsetMap.emplace(static_cast<int>(i) * TEN_PERCENT, 0);
```
Comment (Jojo-Schmitz) [inline src/framework/mpe/mpetypes.h:629]: These lead to many and pretty confusing warnings in MSVC

File: src/notation/notationerrors.h (hunk 2 above)
Comment (Eism) [reply src/notation/notationerrors.h:130]: better fix it as in this commit https://github.com/musescore/MuseScore/pull/10273/commits/e0c96de2e77ac9a6c8222b92606c108da7fb306d

File: src/palette/internal/palette.cpp [hunk 4]
```diff
@@ -64,9 +64,6 @@ Palette::~Palette()
     }
 
     std::vector<PaletteCellPtr> cells = m_cells;
```
Comment (Eism) [inline src/palette/internal/palette.cpp:66]: Please, remove this line too

File: src/notation/notationerrors.h (hunk 2 above)
Comment (Jojo-Schmitz) [reply src/notation/notationerrors.h:130]: OK, done

**Review by Eism — APPROVED** [review]

**Pull request merged**